frame/bounties/src/lib.rs

```diff
@@ -660,12 +785,685 @@ decl_module! {
+										// Give back their deposit.
+										T::Currency::unreserve(
+											&subcurator,
+											subbounty.curator_deposit,
+										);
+										// Continue to change bounty status below...
+									} else if maybe_bounty_active_status
+										.as_ref()
+										.expect("Status is none iff origin is none; qed").0 == sender
+									{
+										// looks like subcurator is inactive,
+										// slash the subcurator deposit.
+										slash_curator(subcurator, &mut subbounty.curator_deposit);
+										// Continue to change bounty status below...
+									} else {
+										// check for expiry
+										// looks like subcurator is inactive,
+										// slash the subcurator deposit.
+										let block_number = system::Pallet::<T>::block_number();
+										if maybe_bounty_active_status
+											.as_ref()
+											.expect("Status is none if origin is none; qed").1 < block_number
+										{
+											slash_curator(
+												subcurator,
+												&mut subbounty.curator_deposit,
+											);
+											// Continue to change bounty status below...
+										} else {
+											// Curator has more time to give an update.
+											return Err(Error::<T>::Premature.into())
+										}
+									}
+								},
+							}
+						},
+						SubBountyStatus::PendingPayout { ref subcurator, .. } => {
+							ensure!(
+								maybe_sender.map_or(
+									true,
+									|sender| maybe_bounty_active_status.map_or(true, |v| v.0 == sender),
+								),
+								BadOrigin,
+							);
+							slash_curator(subcurator, &mut subbounty.curator_deposit);
+							// Continue to change bounty status below...
+						},
+					};
+					// Move the subbounty state to Added.
+					subbounty.status = SubBountyStatus::Added;
+					Ok(())
+				}
+			)?;
+		}
+
+		/// Award subbounty to a beneficiary.
+		///
+		/// The beneficiary will be able to claim the
+		/// funds after a delay.
+		///
+		/// The dispatch origin for this call must be
+		/// the master curator or subcurator of this subbounty.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Subbounty must be in active state, for
+		/// processing the call. And state of subbounty is
+		/// moved to "PendingPayout" on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		/// - `beneficiary`: Beneficiary account.
+		#[weight = <T as Config>::WeightInfo::award_subbounty()]
+		fn award_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+			beneficiary: <T::Lookup as StaticLookup>::Source
+		) {
+			let signer = ensure_signed(origin)?;
+			let beneficiary = T::Lookup::lookup(beneficiary)?;
+
+			// Ensure parent bounty exist,
+			let _ = Self::ensure_bounty_exist(bounty_id, false)?;
+
+			// Ensure subbounty is in expected state
+			SubBounties::<T>::try_mutate_exists(
```

Comment:
why are you using `try_mutate_exists` if you never intend to delete the sub-bounty here?

Reply:
(same concern probably exists in other places as well.)

Reply:
I agree, For implementation of this PR I took the pattern are approach followed in `fn xxx_bounty()`.